packages/sdk-react/src/components/receivables/InvoiceDetails/CreateReceivable/sections/components/ReminderSection/hooks/useValidateCounterpart.ts

```diff
@@ -0,0 +1,55 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+
+import { isIndividualCounterpart } from '@/components/counterparts/helpers';
+import { useMoniteContext } from '@/core/context/MoniteContext';
+import { useCounterpartById } from '@/core/queries';
+
+interface UseValidateCounterpartResult {
+  isEmailValid: boolean;
+  areRemindersEnabled: boolean;
+}
+
+export const useValidateCounterpart = (): UseValidateCounterpartResult => {
+  const { getValues } = useFormContext();
+  const { api } = useMoniteContext();
+
+  const [isEmailValid, setIsEmailValid] = useState(false);
+  const [areRemindersEnabled, setAreRemindersEnabled] = useState(true);
+
+  const counterpartId = getValues('counterpart_id'); //ToDo: Adjust this to match the field name of counterpart Form
+
+  const counterpartQuery = useCounterpartById(counterpartId);
+  const { data: counterpart } = counterpartQuery;
+
+  const validateEmailAndReminders = useCallback(async () => {
+    if (!counterpartId || !counterpartQuery.data || !counterpart) return;
```

Comment:
In my opinion, too many unnecessary checks just increase the required coverage level, while not really necessary.

Reply:
Done, simplified condition